torchtune/modules/rlhf/_generation.py

```diff
@@ -0,0 +1,172 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
```

Comment:
This looks as though it could be not in rlhf, but rather in the general `_generation.py`

Reply:
I could merge the functionality into the existing generation code with sensible defaults, and a `return_logits` flag? 

Though, not sure how you guys feel about return types being `Union[Tuple[..], Tensor]`, but it always feels weird to me to have function calls needing to be unpacked in some places, but not others.